src/main/java/uk/gov/hmcts/reform/slc/services/SendLetterClient.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.reform.slc.services;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.http.client.utils.URIBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URISyntaxException;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+@Component
+public class SendLetterClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(SendLetterService.class);
+
+    private final RestTemplate restTemplate;
+    private final String sendLetterProducerUrl;
+    private final Supplier<ZonedDateTime> currentDateTimeSupplier;
+
+    public SendLetterClient(
+        RestTemplate restTemplate,
+        @Value("${sendletter.producer.url}") String sendLetterProducerUrl,
+        Supplier<ZonedDateTime> currentDateTimeSupplier
+    ) {
+        this.restTemplate = restTemplate;
+        this.sendLetterProducerUrl = sendLetterProducerUrl;
+        this.currentDateTimeSupplier = currentDateTimeSupplier;
+    }
+
+    public void updateSentToPrintAt(UUID letterId) {
+        try {
+            restTemplate.put(normalizedUrl(letterId),
+                ImmutableMap.of(
+                    "sent_to_print_at",
+                    currentDateTimeSupplier.get().format(DateTimeFormatter.ISO_INSTANT)));
+        } catch (URISyntaxException | RestClientException exception) {
```

Comment:
I think this could be simplified quite a bit by 'normalizing' the `sendLetterProducerUrl` once, in the constructor.

Reply:
I don't think this will be of much use if we do it in constructor to be honest.Exception still needs to be throw even if we normalise `sendLetterProducerUrl` in the constructor as the uribuilder build method throws `URISyntaxException` exception.I would say we don't want to complicate this as everyone else in the code no one is validating the uri to be configured :)

Reply:
I'm suggesting not using `URIBuilder` here anymore